### ISSUE TITLE
Status of packaging for dool

### ISSUE THREAD

**Issue by aijanai** [scottchiefbaker/dool#4]
##### SUMMARY

Hi,

I'd like to inquire about the current status and ongoing efforts for dool packaging and distribution.

dool seems to offer snap and rpm packages, but is not present in any of the major Unix distributions' repos.

We might want to elect some maintainers for:
* Mac brew
* Ubuntu
* Centos/RHEL
* Python pip

Anything going on or contributions accepted by the author?

Thanks

**Comment by curoky** [comment]
https://github.com/Homebrew/homebrew-core/pull/87693
I am writing homebrew formula for dool, but it hasn't released yet. https://github.com/scottchiefbaker/dool/issues/4

**Comment by pmoranga** [comment]
Where can I find the RPM package built from it ?

**Comment by scottchiefbaker** [comment]
This is a great idea. I'd love to see `dool` packaged for more distros, and it should be very simple. I just updated the `.spec` file and it now builds correctly (I think) on my Rocky 8 system. I would accept any PRs for package maintainers.

Packaging `dool` should be simple and you can follow the flow in the [install.py](https://github.com/scottchiefbaker/dool/blob/master/install.py) script. 

1. The `dool` script itself needs to go in your `$PATH`
2. Plug-ins go in either `/usr/share/dool/` or `~/.dool/`
3. The man page should go in `/usr/share/man1/`

**Comment by raylu** [comment]
haha I wrote a brew formula but dool doesn't work on macOS because it doesn't have `/proc`. leaving it here for posterity in case dool one day does
```ruby
class Dool < Formula
  desc "Monitor your system"
  homepage "https://github.com/scottchiefbaker/dool"
  url "https://github.com/scottchiefbaker/dool/archive/refs/tags/v1.3.0.tar.gz"
  sha256 "a1537c27e62d12d00d79f60bec456e6a2ec4a0189d44a28ac1ed9effdcb0ec2d"
  license "GPL-3.0-or-later"
  depends_on "python"

  def install
    bin.install "dool"
    man1.install "docs/dool.1"
    pkgshare.install Dir["plugins/*.py"]
  end
end
```

**Comment by anarcat** [comment]
debian is tracked in https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1032875

**Comment by scottchiefbaker** [comment]
@anarcat that's good to know I hadn't seen that yet. Looks like Debian needs to be poked/updated as Dool is very mature now and well supported.

**Comment by scottchiefbaker** [comment]
@aijanai do you have any experience packaging things? I'd love to see this packages for Redhat and Debian but I don't have much experience with that.

**Comment by scottchiefbaker** [comment]
Thanks to Uroš Knupleš we have preliminary `deb` support in #81 now

**Comment by aijanai** [comment]
> @aijanai do you have any experience packaging things? I'd love to see this packages for Redhat and Debian but I don't have much experience with that.

Saw now. Yeah, I previously had experience with both systems. I can give it a try

**Comment by scottchiefbaker** [comment]
@aijanai we currently have packaging for: RHEL (unofficial), Python/PIP, and Debian. As a primarily RedHat user I'd love to get `dool` into the official RedHat repositories but I don't know enough about packaging Python applications to make it happen.

**Comment by iwikus** [comment]
Where is Debian package available? I would like to to download and test it.

**Comment by scottchiefbaker** [comment]
@uroskn did most (all) of the Debian work? Let's loop him in here and see if we can't get packaging stuff back on track. I kind of broke things moving some files around.

**Comment by scottchiefbaker** [comment]
I just landed a `build.sh` script for Debian. It works and builds a `.deb` file but I need testers. Can anyone assist in testing?

Just run `packaging/debian/build.sh` and try installing the resulting `.deb` file

**Comment by scottchiefbaker** [comment]
Here is the file I built: [dool-1.3.4.deb.gz](https://github.com/user-attachments/files/18365221/dool-1.3.4.deb.gz)

Github doesn't allow `.deb` files in issues so I had to gzip it.

**Comment by popey** [comment]
@scottchiefbaker that package installed the files in the root of the filesystem :( 

I now have `/dool`, `/README.md` and friends. I'd remove that deb until it's more widely tested. :)

**Comment by scottchiefbaker** [comment]
I know very little about `.deb` packaging. Need someone who is more familar with it to take a look at our build script.

**Comment by jubalh** [comment]
JFYI: dool is now packaged for [openSUSE](https://build.opensuse.org/package/show/server:monitoring/dool)

**Comment by scottchiefbaker** [comment]
@jubalh thanks for the info on openSUSE. I see some changes were made to the `.spec` file in the openSUSE release. Did you make those changes? We're looking for someone more knowledgeable than me to work on the `.spec` for other OSs as well.

https://bugzilla.redhat.com/show_bug.cgi?id=2263031

**Comment by jubalh** [comment]
@scottchiefbaker yes I maintain dool (and used to maintain dstat) in openSUSE. Our spec file can be seen here: https://build.opensuse.org/projects/server:monitoring/packages/dool/files/dool.spec?expand=1

Different distribution might use different macros in their spec file though. You should be able to take ours and then it might be that you need to make small adjustments to fit them for another distribution.

If you want to push dool to Fedora then you will need to find a maintainer who is doing this. I can help writing a spec file but I'm not Fedora maintainer. Which contains more than just writing a spec file. Need to know about all their processes and rules and have accounts etc.

You could also just write your own spec file and publish an rpm here on GitHub.

**Comment by scottchiefbaker** [comment]
We have an old spec file from dstat that builds: https://github.com/scottchiefbaker/dool/blob/next/packaging/rpm/dool.spec

I don't know much beyond that though. As you mention we need to find a Fedora maintainer as well. As a Redhat user I'd love to see `dool` in Fedora/RHEL so if you know anyone who can help, let me know.

**Comment by jubalh** [comment]
JFYI here you can see where dool is already packaged: https://repology.org/project/dool/versions

**Comment by UrbanDesimator** [comment]
Bit late to the party was trying to catch some fish,  I use dool for monitoring large package builds. Saving the output via tee to compare build info.
Slackware has a package via SBo details below.
https://slackbuilds.org/repository/15.0/system/dool/
